src/contracts/pods/EigenPodManager.sol

```diff
@@ -232,6 +233,7 @@ contract EigenPodManager is
 
     function _addShares(address staker, uint256 shares) internal {
         require(staker != address(0), InputAddressZero());
+        require(int256(shares) >= 0, SharesNegative());
```

Comment:
Noting the equality path where shares passed in could be 0 both for `addShares` and `withdrawSharesAsTokens`.  
It's an edge case where you are completely slashed and your shares withdrawal to complete amount also becomes 0.

In `StrategyManager._addShares` it is currently reverting if the shares amount is 0 (perhaps this is/isn't desired?) but here it will not revert. We should try to have some consistency here and figure out whether we want withdrawals that are completely slashed to revert or complete with amount 0 anyways.

Reply:
Good call out. I've made the behavior consistent with SM: https://github.com/Layr-Labs/eigenlayer-contracts/pull/857/commits/9b9b63645a2647c8f9723d5731d3701005a953f7